extensions/amp-vimeo/0.1/amp-vimeo.js

```diff
@@ -13,66 +13,324 @@
-    iframe.setAttribute('allowfullscreen', 'true');
-    iframe.src = 'https://player.vimeo.com/video/' + encodeURIComponent(
-        videoid);
+    const iframe =
+        htmlFor(element)`<iframe frameborder=0 alllowfullscreen></iframe>`;
+
+    iframe.src = `https://player.vimeo.com/video/${encodeURIComponent(vidId)}`;
+
     this.applyFillContent(iframe);
-    this.element.appendChild(iframe);
+    element.appendChild(iframe);
+
     this.iframe_ = iframe;
+    this.unlistenFrame_ = listen(this.win, 'message', this.boundOnMessage_);
+
+    this.sendCommand_('ping');
+
     return this.loadPromise(iframe);
   }
 
+  /** @override */
+  unlayoutCallback() {
+    this.removeIframe_();
+    return true; // layout again.
+  }
+
+  /** @private */
+  removeIframe_() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenFrame_) {
+      this.unlistenFrame_();
+      this.unlistenFrame_ = null;
+    }
+  }
+
+  /** @private */
+  onReady_() {
+    const {element} = this;
+
+    Object.keys(VIMEO_EVENTS).forEach(event => {
+      this.sendCommand_('addEventListener', event);
+    });
+
+    Services.videoManagerForDoc(element).register(this);
+
+    element.dispatchCustomEvent(VideoEvents.LOAD);
+  }
+
+  /**
+   * @param {!Event} event
+   * @private
+   */
+  onMessage_(event) {
+    if (!isVimeoUrl(event.origin)) {
+      return;
+    }
+
+    if (event.source !== this.iframe_.contentWindow) {
+      return;
+    }
+
+    const eventData = getData(event);
+
+    if (!isJsonOrObj(eventData)) {
+      return;
+    }
+
+    const data = /** @type {?JsonObject} */ (
+      isObject(eventData) ? eventData : tryParseJson(eventData));
+
+    if (data['event'] == 'ready' || data['method'] == 'ping') {
+      this.onReady_();
+      return;
+    }
+
+    const {element} = this;
+
+    const eventToDispatch = VIMEO_EVENTS[data['event']];
+    if (eventToDispatch) {
+      element.dispatchCustomEvent(eventToDispatch);
+    }
+  }
+
   /** @override */
   pauseCallback() {
-    if (this.iframe_ && this.iframe_.contentWindow) {
-      // See
-      // https://developer.vimeo.com/player/js-api
-      this.iframe_.contentWindow./*OK*/postMessage(JSON.stringify(dict({
-        'method': 'pause',
-        'value': '',
-      })), '*');
+    this.pause();
+  }
+
+  /** @override */
+  pause() {
+    this.sendCommand_('pause');
+  }
+
+  /** @override */
+  play() {
+    this.sendCommand_('play');
+  }
+
+  /** @override */
+  mute() {
+    this.sendCommand_(this.setVolumeMethod_(), '0');
+  }
+
+  /** @override */
+  unmute() {
+    // TODO(alanorozco): Set based on volume before unmuting.
+    this.sendCommand_(this.setVolumeMethod_(), '1');
+  }
+
+  /** @override */
+  isInteractive() {
+    return true;
+  }
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  preimplementsMediaSessionAPI() {
+    // TODO(alanorozco): dis tru?
+    return false;
```

Comment:
it is. just checked, they hook into media session API themselves.

Reply:
Fixed. Forgot I left that comment 😛